src/arcam/fmj/state.py

```diff
@@ -115,7 +116,7 @@ def revision(self):
             return self._amxduet.device_revision
         return None
 
-    def get_rc5code(self, table: Dict[Tuple[ApiModel, int], Dict[_T, bytes]], value: _T) -> bytes:
+    def get_code_from_dict(self, table: Dict[Tuple[ApiModel, int], Dict[_T, bytes]], value: _T) -> bytes:
```

Comment:
updated name here since we aren't necessarily passing an RC5 mapping anymore

Reply:
Don't mix these two features. The power mode is a completely fixed feature. It needs no lookup. Just add a map that defines which api models support powermode directly. You only need to list the ones that do.

POWER_WRITE_SUPPORTED = {ApiModel.APISA_SERIES}


if api_model in POWER_WRITE_SUPPORTED:
  do power way
else:
  do rc5 way.

Reply:
done, thanks